File: Frontend/src/app/core/dtos/news.dto.ts
export interface NewsDto {
  id: number;
  title: string;
  content: string;
  summary: string;
  author: string;
  category: string;
  status: 'DRAFT' | 'PUBLISHED' | 'ARCHIVED';
  featured_image: string;
  views: number;
  published_at: string;
  created_at: string;
  updated_at: string;
  facebook_post_id?: string;
  facebook_scheduled_at?: string;
}

export interface NewsListResponse {
  news_list: NewsDto[];
  total_pages: number;
}

export interface NewsCreateRequest {
  title: string;
  content: string;
  summary: string;
  author: string;
  category: string;
  status: 'DRAFT' | 'PUBLISHED' | 'ARCHIVED';
  featured_image: string;
  share_to_facebook?: boolean;
  facebook_scheduled_time?: number;
}

File: Backend/src/main/java/com/example/Sneakers/dtos/NewsDTO.java
package com.example.Sneakers.dtos;

import com.example.Sneakers.models.NewsStatus;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class NewsDTO {
    @JsonProperty("id")
    private Long id;

    @JsonProperty("title")
    @NotBlank(message = "Title is required")
    private String title;

    @JsonProperty("content")
    private String content;

    @JsonProperty("summary")
    private String summary;

    @JsonProperty("author")
    private String author;

    @JsonProperty("category")
    private String category;

    @JsonProperty("status")
    @NotNull(message = "Status is required")
    private NewsStatus status;

    @JsonProperty("featured_image")
    private String featuredImage;

    @JsonProperty("views")
    private Long views;

    @JsonProperty("published_at")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime publishedAt;

    @JsonProperty("created_at")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;

    @JsonProperty("updated_at")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updatedAt;
}

